board/models.py

```diff
@@ -1,3 +1,96 @@
+from django.contrib.auth.models import User
 from django.db import models
 
 # Create your models here.
+
+
+class Board(models.Model):
+    """
+    A database model representing a board, where posts can be created and viewed.
+
+    This model uses a `BinaryField` as opposed to an `ImageField` to store the background image of
+    the board. While this is considered poor design in many cases, Heroku (the cloud provider
+    to be used)'s ephemeral file system renders the file system impossible to use for dynamic
+    storage. The is an issue because `ImageField`, which is a subclass of `FileField`, defaults to
+    storing files on the file system under `MEDIA_ROOT`.
+
+    As a workaround, this model stores the image as a BLOB in the actual database. While the
+    database may inflate as a result, the preservation of the images is guaranteed.
+
+    This board stores a many-to-many relationship to `django.contrib.auth.models.User`, the
+    default `User` model generated by Django. Since Django offers a `ManyToManyField` as a field,
+    an explicit lookup table is not necessary. Instead, users can be added to a board's
+    `admin_users` field, and Django can use lookups across relationships to find all admin users for
+    one specific board. Provided is an example:
+
+    ```
+    # generate and save mock users and boards
+    >>> u1 = User(username="shari", password="i love you") 
+    >>> u1.save() # important! Save to generate an id
+    >>> b1 = Board(name="shari fan club", description="a fan club for shari"...)
+    >>> b1.save()
+
+    # add u1 as admin user to b1
+    >>> b1.admin_users.add(u1)
+
+    # examine relationships
+    >>> b1.admin_users.all()
+    <QuerySet [<User: shari>]>
+    >>> u1.board_set.all()
+    <QuerySet [<Board: shari fan club>]>
+
+    # find all boards where user 1 is an admin (lookup across relationship)
+    >>> Board.objects.filter(admin_users__id=1) 
+    <QuerySet [<Board: shari fan club>]>
+    ```
+
+    Further examples can be found at:
+    https://docs.djangoproject.com/en/3.2/topics/db/examples/many_to_many/
+
+    Class Attributes
+        title -> `CharField`: A charfield with max length 100 containing the title of the board.
+        description -> `CharField`: A charfield with max length 500 containing the title of the board.
+        bg -> `BinaryField`: A field with the background image BLOB.
+        admin_users -> `ManyToManyField`: A field storing the many admin users of this board.
+    """
+    # TODO: look into if we need a lookup table.
+
+    title = models.CharField(max_length=100)
+    description = models.CharField(max_length=200)
+    bg = models.BinaryField()
+    admin_users = models.ManyToManyField(User)
```

Comment:
Can you add another field including the unique hash to be attached after the title? I mean if postgresql pk is already a hash then u can just attach it after the title, but just make sure that it's always a unique hash and not an incremental number ( so people can't guess the link). And also because of how we're allowing same titles so unique hash will be useful. Perhaps you can also add the hash at the end of the `__str__()`.